alpaca/data/historical/corporate_actions.py

```diff
@@ -0,0 +1,140 @@
+            sandbox=False,
+            raw_data=raw_data,
+        )
+
+    def get_corporate_actions(
+        self, request_params: CorporateActionsRequest
+    ) -> Union[RawData, CorporateActionsSet]:
+        """Returns corporate actions data
+
+        Args:
+            request_params (CorporateActionsRequest): The request params to filter the corporate actions data
+        """
+        params = request_params.to_request_fields()
+
+        if request_params.symbols is not None and len(request_params.symbols) > 0:
+            params["symbols"] = ",".join(request_params.symbols)
+        if request_params.types is not None and len(request_params.types) > 0:
+            params["types"] = ",".join(request_params.types)
+
+        response = self._data_get(
+            path="/corporate-actions", api_version=self._api_version, **params
+        )
+        if self._use_raw_data:
+            return response
+
+        return CorporateActionsSet(response)
+
+    def _data_get(
+        self,
+        path: str,
+        limit: Optional[int] = None,
+        page_limit: int = 1000,
+        api_version: str = "v1beta1",
+        **kwargs,
+    ) -> RawData:
+        """Performs Data API GET requests accounting for pagination. Data in responses are limited to the page_limit,
+        which defaults to 1,000 items. If any more data is requested, the data will be paginated.
+
+        Args:
+            limit (Optional[int]): The maximum number of items to query. Defaults to None.
+            page_limit (Optional[int]): The maximum number of items returned per page - different from limit. Defaults to 1000.
+
+        Returns:
+            RawData: Raw Market data from API
+        """
+        params = kwargs
+
+        # data_by_type is in format of
+        #    {
+        #       "type1": [ "data1", "data2", ... ],
+        #       "type2": [ "data1", "data2", ... ],
+        #                ....
+        #    }
+        data_by_type = defaultdict(list)
+
+        total_items = 0
+        page_token = None
+
+        while True:
+            actual_limit = None
+
+            # adjusts the limit parameter value if it is over the page_limit
+            if limit:
+                # actual_limit is the adjusted total number of items to query per request
+                actual_limit = min(int(limit) - total_items, page_limit)
+                if actual_limit < 1:
+                    break
+
+            params["limit"] = actual_limit
+            params["page_token"] = page_token
+
+            response = self.get(path=path, data=params, api_version=api_version)
+
+            d = get_data_from_response(response)
+            [
+                (
+                    data_by_type[ctype].extend(data)
+                    if isinstance(data, list)
+                    else data_by_type[ctype].append(data)
+                )
+                for ctype, data in d.items()
+            ]
```

Comment:
This looks really strange and unpythonic. Please rewrite it to a simple for loop.